R/intervals_support_funs.R

```diff
@@ -0,0 +1,244 @@
+    stop("'data' object must be a PKNCAdata object with 'intervals' and 'data' components.")
+  }
+
+  if (!is.character(target_impute)) {
+    stop("'target_impute' must be a character string.")
+  }
+
+  # Get all parameter column names in the PKNCAdata object
+  all_param_options <- names(sapply(PKNCA.options()$single.dose.aucs, is.logical))
+  logical_cols <- names(which(colSums(data$intervals[sapply(data$intervals, is.logical)]) > 1))
+  param_cols <- intersect(logical_cols, all_param_options)
+
+  # Handle target_params
+  if (is.null(target_params)) {
+    # Take all logical columns in data$intervals that are known parameters
+    target_params <- param_cols
+  } else {
+    # Check that all target_params are logical columns in data$intervals and known parameters
+    missing_params <- setdiff(target_params, param_cols)
+    if (length(missing_params) > 0) {
+      stop("The following target_params are not interval columns and/or known PKNCA parameters: ", paste(missing_params, collapse = ", "))
+      target_params <- intersect(target_params, param_cols)
+    }
+  }
+
+  # Determine the name of the impute column
+  impute_col <- if (!is.na(data$impute)) {
+    data$impute
+  } else if ("impute" %in% colnames(data$intervals)) {
+    "impute"
+  } else {
+    stop("The 'data$intervals' object must contain an impute column either defined in the PKNCAdata object or called `impute`.")
+  }
+
+  # Identify the targeted intervals to which the action is applied
+  mask_target_rows <- data$intervals %>%
+    mutate(
+      is.in.groups = if (!is.null(target_groups)) rowSums(across(all_of(names(target_groups)), ~ . %in% target_groups)) == length(target_groups) else TRUE,
+      is.in.params = rowSums(across(any_of(target_params), ~ . == TRUE)) > 0,
+      is.in.impute = grepl(
+        pattern = paste0(".*(", paste0(target_impute, collapse = ")|("), ").*"),
+        .data[[impute_col]]
+      ),
+      target_rows = is.in.groups & is.in.params & is.in.impute
+    ) %>%
+    pull(target_rows)
+
+  # Create the new version intervals for the target parameters
+  new_intervals_without_impute <- data$intervals %>%
+    filter(mask_target_rows) %>%
+    mutate(across(any_of(param_cols), ~FALSE)) %>%
+    mutate(across(any_of(target_params), ~TRUE)) %>%
+    rowwise() %>%
+    mutate(!!impute_col := paste0(setdiff(unlist(strsplit(.data[[impute_col]], ",")), target_impute),
+      collapse = ","
+    )) %>%
+    mutate(!!impute_col := ifelse(.data[[impute_col]] == "", NA_character_, .data[[impute_col]])) %>%
+    ungroup() %>%
+    as.data.frame()
+
+  # Make parameters FALSE in target intervals
+  data$intervals[mask_target_rows, target_params] <- FALSE
+
+  # Combine and remove intervals where all logical parameter columns are FALSE
+  data$intervals <- rbind(data$intervals, new_intervals_without_impute) %>%
+    filter(rowSums(across(any_of(param_cols), as.numeric)) > 0)
+
+  return(data)
```

Comment:
Please omit the `return()` part and have the return value use the implicit return method.